packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -444,7 +444,7 @@ class HotRunner extends ResidentRunner {
               final ServiceEvent pauseEvent = view.uiIsolate.pauseEvent;
               if ((pauseEvent != null) && pauseEvent.isPauseEvent) {
                 // Resume the isolate so that it can be killed by the embedder.
-                return view.uiIsolate.resume();
```

Comment:
This seems wrong

Reply:
Let me investigate what this code is supposed to do ...

Reply:
Yeah this one's super weird. AFAICT, there is no reason `then(THIS)` should have to return something. I think the error,

```
This function has a return type of 'Future<Map<String, dynamic>>', but doesn't
end with a return statement • packages/flutter_tools/lib/src/run_hot.dart:443:13
```

is saying that one path returns a `Future<Map<String, dynamic>>`, and another pass doesn't return anything, and those two paths then don't match any consistent return type. This return value isn't used anywhere, right?

Reply:
The whenComplete triggers off of the returned future, or the immediate result of `return null`. in this case it should be:

```
if ((pauseEvent != null) && pauseEvent.isPauseEvent) {
  return view.uiIsolate.resume();
}
return null;
```
```

Reply:
Done!